### PR TITLE
Add test to show some issues with custom scale

### PR DESCRIPTION
**Pull request by james-rabbit**
I found some issues when using custom scale, including an infinite loop that causes human-format to hang forever.
This PR adds a test case for the three problems I encountered, with comments for how they failed.

### CODE REVIEW

File: index.spec.js [hunk 1]
```diff
@@ -74,6 +74,19 @@ describe('humanFormat()', function () {
       })
     })
 
+    it('should use a simplified custom scale', function () {
+      var scale = new humanFormat.Scale({
+        K: 1000,
+        M: 1000000,
+        B: 1000000000
+      })
+      expect(humanFormat(100, { scale: scale, decimals: 0 })).toBe('100') // fails, this returns "0 K" and there is no way to specify no prefix
```
Comment (julien-f) [inline index.spec.js]: Not a bug, you need to add an empty prefix:

```js
var scale = new humanFormat.Scale({
  '': 1,
  'K': 1e3,
  'M': 1e6,
  'B': 1e9,
})
```

Though I agree it's not very obvious 😛

File: index.spec.js [hunk 2]
```diff
@@ -74,6 +74,19 @@ describe('humanFormat()', function () {
...
+      expect(humanFormat(1000, { scale: scale, decimals: 0 })).toBe('1 K')
+      expect(humanFormat(1000000, { scale: scale, decimals: 0 })).toBe('1 M')
+      expect(humanFormat(1000000000, { scale: scale, decimals: 0 })).toBe('1 B') // runs infinitely in the do-while loops
+      expect(humanFormat(1000000000000, { scale: scale, decimals: 0 })).toBe('1,000 B') // fails, this returns "1000 B"
```
Comment (julien-f) [inline index.spec.js]: Not a bug either, it currently does not format the number according to the locale, it simply converts the number to string.

I'm not against a PR if you need this though.

File: index.spec.js [hunk 3]
```diff
@@ -74,6 +74,19 @@ describe('humanFormat()', function () {
...
+    it('should use a simplified custom scale', function () {
+      var scale = new humanFormat.Scale({
+        K: 1000,
+        M: 1000000,
+        B: 1000000000
+      })
+      expect(humanFormat(100, { scale: scale, decimals: 0 })).toBe('100') // fails, this returns "0 K" and there is no way to specify no prefix
+      expect(humanFormat(1000, { scale: scale, decimals: 0 })).toBe('1 K')
+      expect(humanFormat(1000000, { scale: scale, decimals: 0 })).toBe('1 M')
+      expect(humanFormat(1000000000, { scale: scale, decimals: 0 })).toBe('1 B') // runs infinitely in the do-while loops
```
Comment (julien-f) [inline index.spec.js]: Definitely sounds like a bug, I'll take a look.

**Comment by james-rabbit** [comment]
Thanks!

**Pull request closed without merging**